TLM/TLM/UI/Translation.cs

```diff
@@ -12,77 +17,254 @@ namespace TrafficManager.UI {
     using Util;
 
     public class Translation {
-        public static readonly IDictionary<string, string> LanguageLabels;
-        public static readonly IList<string> AvailableLanguageCodes;
+        /// <summary>
+        /// Defines the order of language selector dropdown.
+        /// </summary>
+        public static readonly IList<string> AvailableLanguageCodes
+            = new List<string> {
+                                   "en",
+                                   "en-gb",
+                                   "de",
+                                   "es",
+                                   "fr",
+                                   "hu",
+                                   "it",
+                                   "ja",
+                                   "ko",
+                                   "nl",
+                                   "pl",
+                                   "pt",
+                                   "ru",
+                                   "zh", // Mainland China (simplified)
+                                   "zh-tw", // Taiwanese (traditional)
+                               };
+
+        /// <summary>
+        /// Defines column order in CSV, but not the dropdown order.
+        /// Mapping from translation languages (first row of CSV export) to language locales used in
+        /// <see cref="AvailableLanguageCodes"/> above.
+        /// </summary>
+        private static readonly Dictionary<string, string> csv_columns_to_locales_
+            = new Dictionary<string, string>() {
+                                                   { "English", DEFAULT_LANGUAGE_CODE },
+                                                   { "Chinese Simplified", "zh" },
```

Comment:
We also need a lookup of display language too (what to display in the drop-down in mod options). It looks like this is going to make all the language names in that list English?

Specifically, this sort of thing:

```csharp
            LanguageLabels = new TinyDictionary<string, string> {	
                ["de"] = "Deutsch",	
                ["en"] = "English",	
                ["es"] = "Español",	
                ["fr"] = "Français",	
                ["hu"] = "Magyar",	
                ["it"] = "Italiano",	
                ["ja"] = "日本語",	
                ["ko"] = "한국의",	
                ["nl"] = "Nederlands",	
                ["pl"] = "Polski",	
                ["pt"] = "Português",	
                ["ru"] = "Русский язык",	
                ["zh-tw"] = "中文 (繁體)",	
                ["zh"] = "中文 (简体)"	
            };
```

Reply:
No, this is column names in CSV (language names set by Crowdin go into column names in Excel/CSV)
The actual language names are moved into the translations.